plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -375,25 +411,80 @@ public void setViewPrincipal(ConnectorSession session, SchemaTableName schemaVie
+        getViews(session, namespace).keySet().forEach(views::add);
+        return views.build();
     }
 
     @Override
     public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, Optional<String> namespace)
     {
-        return ImmutableMap.of();
+        List<Namespace> namespaces = relevantNamespaces(session, namespace);
+        SessionContext sessionContext = convert(session);
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
+        for (Namespace restNamespace : namespaces) {
+            List<TableIdentifier> restViews = restSessionCatalog.listViews(sessionContext, restNamespace);
+            for (TableIdentifier restView : restViews) {
+                SchemaTableName schemaTableName = SchemaTableName.schemaTableName(restView.namespace().toString(), restView.name());
+                getView(session, schemaTableName).ifPresent(view -> views.put(schemaTableName, view));
+            }
+        }
+
+        return views.buildOrThrow();
     }
 
     @Override
     public Optional<ConnectorViewDefinition> getView(ConnectorSession session, SchemaTableName viewName)
     {
-        return Optional.empty();
+        Optional<View> icebergView = getIcebergView(session, viewName);
+        return icebergView.map(this::resolveViewDefinition);
+    }
+
+    private ConnectorViewDefinition resolveViewDefinition(View view)
+    {
+        Optional<String> comment = Optional.ofNullable(view.properties().get(ICEBERG_VIEW_COMMENT));
+        ViewVersion currentVersion = view.currentVersion();
+        List<ConnectorViewDefinition.ViewColumn> viewColumns = IcebergUtil.viewColumnsFromSchema(typeManager, view.schema());
+        ConnectorViewDefinition viewDefinition = null;
+        for (ViewRepresentation representation : currentVersion.representations()) {
+            if (representation instanceof SQLViewRepresentation sqlView) {
+                Optional<String> catalog = Optional.ofNullable(currentVersion.defaultCatalog());
+                Optional<String> schema = Optional.empty();
+                if (!currentVersion.defaultNamespace().isEmpty()) {
+                    schema = Optional.of(currentVersion.defaultNamespace().toString());
+                }
+
+                // Found a trino dialect view, so that should get used
+                if (sqlView.dialect().equals(ICEBERG_VIEW_SQL_DIALECT)) {
+                    Optional<String> owner = Optional.ofNullable(view.properties().get(ICEBERG_VIEW_OWNER));
+                    boolean runAsInvoker = Boolean.parseBoolean(view.properties().get(ICEBERG_VIEW_RUN_AS_INVOKER));
+                    return new ConnectorViewDefinition(
+                            sqlView.sql(), catalog, schema, viewColumns, comment, owner, runAsInvoker, null);
+                }
+                else {
+                    // Set a best effort view definition
+                    viewDefinition = new ConnectorViewDefinition(
```

Comment:
This could be tested with a product test 

see io.trino.tests.product.iceberg.TestIcebergSparkCompatibility and `ICEBERG_REST`  product test group for reference.

Reply:
@findinpath yes, cross-engine view support can be tested with a product test
in this PR it's gonna be removed though, see https://github.com/trinodb/trino/pull/19818#discussion_r1409256153